langlib/lang.table/src/main/ballerina/table.bal

```diff
@@ -160,7 +166,7 @@ public isolated function reduce(table<Type> t, @isolatedParam function(Type1 acc
 # + t - the table
 # + k - the key
 # + return - the member of parameter `t` that had key parameter `k`
-public isolated function remove(table<Type> key<KeyType> t, KeyType k) returns Type = @java:Method {
+public isolated function remove(table<MapType> key<KeyType> t, KeyType k) returns MapType = @java:Method {
```

Comment:
```ballerina
type MapType map<any|error>;
type KeyType anydata;

public function main() {
    table<MapType> key<KeyType> t = table [];
}
```
Specifying a `key-type-constraint` for a `table` constrained with a `map` type results in `table with constraint of type map cannot have key specifier or key type constraint` compilation error. Shouldn't the same be applicable here when such `table` type is given as a function param?

Reply:
```ballerina
function foo(table<MapType> key<KeyType> t) {}
```
This should not be allowed right? This doesn't result in a compilation error ATM.

Reply:
hmm. yeah. I think we will have to handle it specifically, by checking if it is not a langlib function. will send the fix with this pr